app/src/main/java/com/orange/ouds/app/ui/components/topappbar/TopAppBarDemoScreen.kt

```diff
@@ -0,0 +1,208 @@
+import com.orange.ouds.app.ui.components.colorArgument
+import com.orange.ouds.app.ui.components.contentDescriptionArgument
+import com.orange.ouds.app.ui.components.onClickArgument
+import com.orange.ouds.app.ui.components.painterArgument
+import com.orange.ouds.app.ui.utilities.Code
+import com.orange.ouds.app.ui.utilities.LocalThemeDrawableResources
+import com.orange.ouds.app.ui.utilities.ThemeDrawableResources
+import com.orange.ouds.app.ui.utilities.composable.AppPreview
+import com.orange.ouds.app.ui.utilities.composable.CustomizationFilterChips
+import com.orange.ouds.app.ui.utilities.composable.CustomizationSwitchItem
+import com.orange.ouds.app.ui.utilities.composable.CustomizationTextField
+import com.orange.ouds.app.ui.utilities.composable.DemoScreen
+import com.orange.ouds.core.component.OudsCenterAlignedTopAppBar
+import com.orange.ouds.core.component.OudsLargeTopAppBar
+import com.orange.ouds.core.component.OudsMediumTopAppBar
+import com.orange.ouds.core.component.OudsTopAppBar
+import com.orange.ouds.core.component.OudsTopAppBarAction
+import com.orange.ouds.core.component.OudsTopAppBarNavigationIcon
+import com.orange.ouds.core.theme.OudsTheme
+import com.orange.ouds.foundation.extensions.orElse
+import com.orange.ouds.theme.OudsVersion
+
+@Composable
+fun TopAppBarDemoScreen() {
+    val state = rememberTopAppBarDemoState()
+    val themeDrawableResources = LocalThemeDrawableResources.current
+    DemoScreen(
+        description = stringResource(id = Component.TopAppBar.descriptionRes),
+        bottomSheetContent = { TopAppBarDemoBottomSheetContent(state = state) },
+        codeSnippet = { topAppBarDemoCodeSnippet(state = state, themeDrawableResources = themeDrawableResources) },
+        demoContent = { TopAppBarDemoContent(state = state) },
+        demoContentPaddingValues = PaddingValues(horizontal = OudsTheme.spaces.fixed.none),
+        version = OudsVersion.Component.Bar
+    )
+}
+
+@Composable
+private fun TopAppBarDemoBottomSheetContent(state: TopAppBarDemoState) {
+    with(state) {
+        CustomizationFilterChips(
+            applyTopPadding = false,
+            label = stringResource(R.string.app_components_common_size_label),
+            chipLabels = TopAppBarDemoState.Size.entries.map { stringResource(it.labelRes) },
+            selectedChipIndex = TopAppBarDemoState.Size.entries.indexOf(size),
+            onSelectionChange = { id -> size = TopAppBarDemoState.Size.entries[id] }
+        )
+        CustomizationSwitchItem(
+            label = stringResource(R.string.app_components_topAppBar_centerAligned_label),
+            checked = centerAligned,
+            onCheckedChange = { centerAligned = it },
+            enabled = centerAlignedSwitchEnabled
+        )
+        CustomizationSwitchItem(
+            label = stringResource(R.string.app_components_topAppBar_navigationIcon_label),
+            checked = navigationIcon,
+            onCheckedChange = { navigationIcon = it }
+        )
```

Comment:
Should we demonstrate all the predefined icons by using a chips list : "none", "back", "close", "menu"?